frontend/src/metabase/components/tree/TreeNode.styled.tsx

```diff
@@ -36,13 +36,19 @@ export const ExpandToggleButton = styled.button`
   visibility: ${props => (props.hidden ? "hidden" : "visible")};
 `;
 
-interface ExpandToggleIconProps {
+interface ExpandToggleIconProps extends IconProps {
   isExpanded: boolean;
 }
 
-export const ExpandToggleIcon = styled(Icon, {
-  shouldForwardProp: propName => propName !== "isExpanded",
-})<ExpandToggleIconProps & IconProps>`
+export const ExpandToggleIcon = styled(
+  ({ isExpanded, ...props }: ExpandToggleIconProps) => (
```

Comment:
Instead of using `shouldForwardProp`, we can simply prevent `isExpanded` from being spread into `Icon`. I think this is more readable.